File: core/src/main/java/com/lomekwi/cave/util/Ranges.java
package com.lomekwi.cave.util;

import com.google.common.collect.Range;

public final class Ranges {
    private Ranges(){}
    public static Range<Long> shift(Range<Long> r, long delta) {
        return Range.closedOpen(r.lowerEndpoint() + delta, r.upperEndpoint() + delta);
    }
}

File: core/src/main/scala/com/lomekwi/cave/util/Ranges.scala
package com.lomekwi.cave.util

import com.google.common.collect.Range

object Ranges {
  def shift(r: Range[java.lang.Long], delta: Long): Range[java.lang.Long] = {
    Range.closedOpen(r.lowerEndpoint() + delta, r.upperEndpoint() + delta)
  }
}
